### PR TITLE
Remove the 29 byte limitation, when requesting bluetooth devices

### PR DESCRIPTION
**Pull request by zakorgy**
Remove the 29 byte limit when calling `requestDevice` using a `name` or `namePrefix`. The following discussion is about this by the specification writers: https://bugs.chromium.org/p/chromium/issues/detail?id=653718.
Related tests also got removed.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] There are tests for these changes

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15366)

### CODE REVIEW

File: components/script/dom/bluetooth.rs [hunk 1]
```diff
@@ -40,11 +40,6 @@ const MANUFACTURER_DATA_ERROR: &'static str = "'manufacturerData', if present, m
 const MASK_LENGTH_ERROR: &'static str = "`mask`, if present, must have the same length as `dataPrefix`.";
 // 248 is the maximum number of UTF-8 code units in a Bluetooth Device Name.
 const MAX_DEVICE_NAME_LENGTH: usize = 248;
-// A device name can never be longer than 29 bytes.
-// An advertising packet is at most 31 bytes long.
-// The length and identifier of the length field take 2 bytes.
-// That leaves 29 bytes for the name.
-const MAX_FILTER_NAME_LENGTH: usize = 29;
```
Comment (nox) [inline components/script/dom/bluetooth.rs]: The spec just says that the new limit is 248.

> If the UTF-8 encoding of filter.name is more than 248 bytes long, throw a TypeError and abort these steps.

**Comment by zakorgy** [comment]
The 248 limit is also in the current implementation [here](https://github.com/servo/servo/pull/15366/files#diff-c46adaa4a07505ce292afa8642267f23L42) and used as you quoted. The 29 byte limit was not a part of the specification, but when we ported chrome tests, we discovered they use it and we put this check in our code too. We fired an [issue](https://github.com/WebBluetoothCG/web-bluetooth/issues/296), to move this limit in the specification. Meanwhile it turned out we don't need this anymore, this is what the linked conversation about.

**Comment by nox** [comment]
Oh right, misread the code.

@bors-servo r+

**Pull request merged**